exir/tests/test_memory_format_ops_pass_utils.py

```diff
@@ -99,10 +104,57 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         return t1 * t2
 
 
+class AmbiguousDimOrderError(RuntimeError):
+    pass
+
+
+def assert_unambiguous_dim_order(gm):
+    class ExampleNOPPass(ExportPass):
+        """
+        Does nothing!
+        """
+
+        def call_operator(self, op, args, kwargs, meta):
+            return super().call_operator(
+                op,
+                args,
+                kwargs,
+                meta,
+            )
+
+    # This is an example of how one can detect ambiguous dim_order anywhere in the graph.
+    # You can be surgical and only detect it in the nodes you are interested in or something else.
+    def detect_ambiguity(gm):
+        """
+        Check every node's output tensor dim_order and raise if it is ambiguous for a list of formats.
+        """
+        for node in gm.graph.nodes:
+            if node.op == "call_function":
+                tensor = node.meta["val"]
+                if not isinstance(tensor, torch.Tensor):
```

Comment:
Maybe here we need also take care of tensor list?

Reply:
Fixed in [865c7f8](https://github.com/pytorch/executorch/pull/9371/commits/865c7f8b1267ba6d5ad5cbee5ced24c909a33514)